lib/kube/proxy/cluster_details.go

```diff
@@ -159,19 +163,42 @@ func newClusterDetails(ctx context.Context, cfg clusterDetailsConfig) (_ *kubeDe
 		gvkSupportedResources: gvkSupportedRes,
 	}
 
+	// If cluster is online and there's no errors, we refresh details seldom (every 5 minutes),
+	// but if the cluster is offline, we try to refresh details more often to catch it getting back online earlier.
+	firstPeriod := defaultRefreshPeriod
+	if isClusterOffline {
+		firstPeriod = backoffRefreshStep
+	}
+	refreshDelay, err := retryutils.NewLinear(retryutils.LinearConfig{
+		First:  firstPeriod,
+		Step:   backoffRefreshStep,
+		Max:    defaultRefreshPeriod,
+		Jitter: retryutils.NewSeventhJitter(),
+		Clock:  cfg.clock,
+	})
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
```

Comment:
Are you missing a stop call?

Reply:
Added `Close` call https://github.com/gravitational/teleport/pull/43903/commits/34f66da15c6a04b1f1690cb2e1330adf44dd0f54

Reply:
That'll work on error, but what about the happy path? Is there no way to terminate this goroutine?

Reply:
It's terminated by canceling the `ctx` (which also happens on `Close`). I didn't change this part of logic, I only removed the ticker.

Reply:
As Anton said, this is a long-lived routine that is closed once the kube cluster is unregistered from the agent - dynamic kube clusters or during the agent shutdown process. When that happens, close is called and the context is canceled. The close waits until all the goroutines are collected.